### PR TITLE
[SYCL] Reuse plugin information so that it's not copied when sycl::detail::plugin is copied

### PR DESCRIPTION
**Pull request by s-kanaev**

### CODE REVIEW

**Comment by s-kanaev** [comment]
@smaslov-intel , do you any potential caveats here?

File: sycl/source/detail/pi.cpp [hunk 1]
```diff
@@ -378,9 +379,15 @@ static void initializePlugins(std::vector<plugin> &Plugins) {
...
-  PluginInformation.PiFunctionTable = {};
+  std::shared_ptr<PiPlugin> PluginInformation;
+  {
+    PiPlugin PluginInformationInstance{
+        _PI_H_VERSION_STRING, _PI_H_VERSION_STRING, nullptr, {}};
+    PluginInformationInstance.PiFunctionTable = {};
+
+    PluginInformation.reset(new PiPlugin);
+    *PluginInformation = PluginInformationInstance;
+  }
```
Comment (romanovvlad) [inline sycl/source/detail/pi.cpp:390]: Why can't we just allocate `PiPlugin` and assign it to `PluginInformation` directly?
```suggestion
  std::shared_ptr<PiPlugin> PluginInformation{new PiPlugin{_PI_H_VERSION_STRING, _PI_H_VERSION_STRING, /*Targets=*/nullptr, /*FunctionPointers=*/{}};
```

File: sycl/source/detail/pi.cpp [hunk 2]
```diff
@@ -439,8 +446,6 @@ static void initializePlugins(std::vector<plugin> &Plugins) {
   }
 
 #ifdef XPTI_ENABLE_INSTRUMENTATION
-  GlobalHandler::instance().getXPTIRegistry().initializeFrameworkOnce();
```
Comment (romanovvlad) [inline sycl/source/detail/pi.cpp:442]: Could you please tell why this line is being removed?

Comment (s-kanaev) [reply sycl/source/detail/pi.cpp:442]: It's an unwanted change.
Reverted it.

File: sycl/source/detail/pi.cpp (hunk 1 above)
Comment (s-kanaev) [reply sycl/source/detail/pi.cpp]: Changed using `make_shared`.

File: sycl/source/detail/plugin.hpp [hunk 3]
```diff
@@ -236,7 +241,7 @@ class plugin {
   std::shared_ptr<std::mutex> getPluginMutex() { return MPluginMutex; }
 
 private:
-  RT::PiPlugin MPlugin;
+  std::shared_ptr<RT::PiPlugin> MPlugin;
```
Comment (romanovvlad) [inline sycl/source/detail/plugin.hpp:244]: Shouldn't we have one to one mapping between `plugin` and `RT::PiPlugin` and one to many between `plugin` and `platform` ?

Comment (s-kanaev) [reply sycl/source/detail/plugin.hpp:244]: You're right. Though, adding such a change to this patch will make it more complex for review.
Shifting to 1-1 plugin-pi_plugin and 1-n plugin-platform is the next step.

Comment (smaslov-intel) [reply sycl/source/detail/plugin.hpp:244]: Don't we already have 1:many mapping to platforms at line 253?

**Review by smaslov-intel — APPROVED** [review]
LGTM

File: sycl/source/detail/plugin.hpp (hunk 3 above)
Comment (s-kanaev) [reply sycl/source/detail/plugin.hpp:244]: @smaslov-intel , there's such a mapping between `sycl::detail::plugin` and `pi_platform`. Though, I have an intent to add the same mapping for `sycl::detail::plugin` and `sycl::detail::platform_impl`.

**Pull request merged**